src/scenes/Game.js

```diff
@@ -135,6 +135,30 @@ export default class extends Phaser.Scene {
     // set the boundaries of our game world
     this.physics.world.bounds.width = this.groundLayer.width;
     this.physics.world.bounds.height = this.groundLayer.height;
+
+    // Game Timer
+    this.timerText = ""
+    this.timerSeconds = 60 * 2;
+    this.timer = this.time.addEvent({
+      delay: 1000,
+      callback: () => {
+        this.timerSeconds-=1;
+        let seconds = this.timerSeconds % 60;
+        let minutes = Math.floor(this.timerSeconds / 60)
+        seconds = seconds > 9 ? seconds + '' : '0' + seconds
+        minutes = minutes > 9 ? minutes + '' : '0' + minutes
+        this.timerText = `${minutes}:${seconds}`
+        if (minutes === 0 && seconds === 0) {
+          // Reset the timer / scene
```

Comment:
We'll probably need to fade out to another scene with a message and fade back into the new / reset scene. Could be reused for death too (The day is over, You died, etc...)